metricsbp/baseplate_hooks.go

```diff
@@ -101,3 +106,76 @@ var (
 	_ tracing.CreateChildSpanHook  = (*spanHook)(nil)
 	_ tracing.AddSpanCounterHook   = (*spanHook)(nil)
 )
+
+// NewConcurrencyCreateSpanHook returns a CreateServerSpanHook that can be used
+// to monitor the number of concurrent requests being processed and spawns a
+// reporter gorouting to publish that as a gauge.
+//
+// The reporter will update the gauge at the given interval.
+// The reporter goroutine will be stopped when the metrics context is canceled.
+// The gauge will be named "runtime.{hostname}.PID{process_id}.active_requests".
+func NewConcurrencyCreateSpanHook(metrics *Statsd, interval time.Duration) tracing.CreateServerSpanHook {
+	var activeRequests int64
+	counter := &atomicCounter{count: &activeRequests}
+	hostname, err := os.Hostname()
+	if err != nil {
+		hostname = "UNKOWN-HOSTNAME"
+	}
+	gauge := metrics.Gauge(
+		fmt.Sprintf("runtime.%s.PID%d.active_requests", hostname, os.Getpid()),
+	)
+	go func() {
+		for {
+			time.Sleep(interval)
+			if metrics.Ctx().Err() != nil {
+				return
+			}
+			gauge.Set(float64(counter.current()))
+		}
+	}()
+	return &concurrencyCreateServerSpanHook{counter: counter}
+}
+
+type atomicCounter struct {
+	count *int64
+}
```

Comment:
I also feel this is overkill/overcomplex.

That gauge, by its definition, is not per-request but only one per process. So it perfectly fine to just declare a global variable of int64, and just call `atomic.AddInt64` 1/-1 in the hook, and call `atomic.LoadInt64` in `RunSysStats` to report it.

Reply:
I dunno, to me directly editing a global value using the `atomic` functions seems like a bad example to set?  I don't think this is complex at all and sets a better example for other developers reading the code.

I can change it if that's not the case.